mimic/rest/ironic_api.py

```diff
@@ -21,6 +21,20 @@ def __init__(self, core):
         """
         self.core = core
 
+    @app.route('/nodes', methods=['POST'])
+    def create_node(self, request):
+        """
+        Responds with response code 201 and returns the newly created node.
+        """
+        return self.core.ironic_node_store.create_node(request)
+
+    @app.route('/nodes/<string:node_id>', methods=['DELETE'])
+    def delete_node(self, request, node_id):
+        """
+        Responds with response code 204 and delete the node.
+        """
+        return self.core.ironic_node_store.delete_node(request, node_id)
```

Comment:
I agree that it *seems* like there's a layering violation here - and honestly, maybe there is one and we could figure out a better structure - but Mimic is a bit of a special case in terms of the correspondence between its "model" later and its "view" (or public-facing HTTP API layer).  Mimic's models are modeling not just the abstract state of "servers" or "load balancers" or whatever, but *API behaviors* as well, which means that in the "model" layer we occasionally need representations of things like "the backend is down, return a `503` for some reason" or "randomly truncate the HTTP response in the middle, we saw that happen when talking to the load balancer API one time when trying to manipulate this one messed up CLB".  As such, passing along the full HTTP request and all its capabilities seems like the best option available, although I'd love to hear other thoughts, because this definitely feels not quite right based on every *other* project I've ever worked on :).